packages/ui-components/rollup.config.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import json from '@rollup/plugin-json';
+import builtins from 'rollup-plugin-node-builtins';
+import { terser } from 'rollup-plugin-terser';
+import commonjs from '@rollup/plugin-commonjs';
+import analyze from 'rollup-plugin-analyzer';
+import babel from '@rollup/plugin-babel';
+import external from 'rollup-plugin-peer-deps-external';
+import del from 'rollup-plugin-delete';
+import svg from 'rollup-plugin-svg';
+import pkg from './package.json';
+
+const plugins = [
+  builtins(),
+  external(),
+  json({
+    compact: true,
+  }),
+  svg(),
+  babel({
+    exclude: 'node_modules/**',
+    configFile: './.babelrc.js',
+  }),
+  commonjs(),
+  analyze({ summaryOnly: true }),
+];
+
+const externals = [
+  ...Object.keys(pkg.devDependencies || {}),
+  ...Object.keys(pkg.dependencies || {}),
+  /@material-ui/,
+  /date-fns/,
```

Comment:
Why do these two need to be separately listed?

Reply:
Yeah I don't love it either - I'm open to future improvements for sure! Those patters for @material-ui etc are needed to make sure that imports that don't match the name from dependencies exactly are picked up. Eg. `@material-ui/core/button` or `date-fns/locale/la` and so on.

Reply:
Cool makes sense!